include/ipfs/client.h

```diff
@@ -93,6 +93,32 @@ class Client {
        * properties "Repo", "System", "Version". */
       Json* version);
 
+  /** Download an IPFS object
+   *
+   * Implements
+   * https://ipfs.io/docs/api/#api-v0-get
+   *
+   * An example usage:
+   * @snipper generic.cc ipfs::Client::Get
+   *
+   * @throw std::exception if any errors
+   *
+   * @since version 0.2.0 */
+  void Get(
+      /** [in] The path to the IPFS object */
+      const std::string& arg,
+      /** [out] The contents of the object read */
+      std::iostream* contents,
+      /** [in] The path where the output should be stored. If empty will be
+       * ignored */
+      const std::string& output = std::string(),
```

Comment:
What exactly is this argument used for? From reading the comment I would expect the result to be saved in this file, but we don't do that in our implementation of the newly added method Get().

Reply:
The `output` argument is used by the ipfs daemon to determine where to save the file to. It is one of the arguments as described in the spec.

Reply:
Hmm, have you tried this? It does not work for me - no file is created with the specified name. Also, it would be very strange to be able to instruct the ipfs daemon to create files on its local filesystem. It just does not make sense because the daemon may be running on a different machine than the client and then the client will not be able to access the created file. Also, I assume all communication between the ipfs daemon and the client is happening via http, not by planting files on the filesystem.

Another thing I noticed is that always a tar archive is saved in `contents`, regardless of the value of the `archive` argument :/

Reply:
That's really weird since it's conforming to the spec. I'll do some more testing when I get the time then